### PR TITLE
Convert README into Markdown

### PR DESCRIPTION
**Pull request by bobvanderlinden**
As per #4, I converted the README to markdown and added information about the new `HOSTS_FILE` environment variable.

Let me know if the conversion to markdown is desired.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+with functionality that fetches these entries from a user-defined hosts file instead of `/etc/hosts`.
+
+By default `hosts` fetches hosts entries from `~/.hosts`, but `HOSTS_FILE` can be used to specify a different file as well.
+
+## `HOSTALIASES`
+
+Some variants of libc have support for the `HOSTALIASES` environment variable.
+This is similar to using `HOSTS_FILE` in this library.
+However, the variant of glibc used on Debian and ArchLinux, amongst others, do not have this feature.
+
```
Comment (figiel) [inline README.md:10]: Let's remove this whole chapter, I don't think we want to keep this up to date with regards to what's implemented elsewhere.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Using the `HOSTS_FILE` environment variable:
+
+```sh
+echo "127.0.0.1 somehost" >> hosts
+HOSTS_FILE=$PWD/hosts nc -vz somehost 80
+```
+
+## Limitations
+
+`hosts` does not work with suid programs due to suid programs clearing the environment before starting. That includes `ping` unfortunately.
```
Comment (figiel) [inline README.md]: That's not exactly true (it's a feature of loader to ignore LD_PRELOAD in this case), let's drop "due to suid programs clearing the environment before starting".

File: README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# hosts
+
+Allow user-defined hosts entries.
+
+`hosts` is a `LD_PRELOAD` library that overrides the fetching of hostnames in applications
+with functionality that fetches these entries from a user-defined hosts file instead of `/etc/hosts`.
```
Comment (figiel) [inline README.md]: I don't think this is clear or accurate, hostnames aren't fetched, they are resolved to IP addresses. I think more precise description wouldn't hurt if you are doing the rewrite. The audience for this is rather deeply technical anyway.

**Review by figiel — COMMENTED** [review]
I'm fine with conversion to markdown, with the ~rewrite too, but please take a look at the comments.

**Comment by bobvanderlinden** [comment]
Thanks for the comments! I've worked on the readme some more. Incorporated your comments. I also added an example of name resolving redirection (resolve `somehost` using `localhost`, instead of `127.0.0.1`). Let me know what you think :+1:

**Comment by figiel** [comment]
Hi, sorry I can't see the changes compared to the version you sent initially. Did you forget to push?

**Comment by bobvanderlinden** [comment]
Absolutely right :+1: Pushed them now.

**Pull request merged**